### ISSUE TITLE
NVDA doesn't read aria-label inside a link correctly

### ISSUE THREAD

**Issue by swidjaja** [nvaccess/nvda#12024]
### Steps to reproduce:
1. Open the following html in browser (I used Microsoft Edge)
```html
<a href="https://www.google.com">
  <span>Product title</span>
  <span aria-label="Offer price $9.00. Original price $10.00">$9.00</span>
</a>

2. Navigate to the the link above

### Actual behavior:
NVDA announces the link as  "Product title $9.00 link".

### Expected behavior:
NVDA announces the link as  "Product title offer price $9.00. Original price $10.00 link"

### System configuration
#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
NVDA Version 2020.3

#### Windows version:
Windows 10 Enterprise

#### Name and version of other software in use when reproducing the issue:
Microsoft Edge 44.18362.449.0

#### Other information about your system:

### Other questions
#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

#### If addons are disabled, is your problem still occuring?
Unknown

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

**Comment by jcsteh** [comment]
Per the ARIA spec, aria-label is prohibited on a generic element such as a span. See https://w3c.github.io/aria/#generic.
